### PR TITLE
Added option to use files with (word , weight) as described in Issue #226

### PR DESCRIPTION
**Pull request by mohammadKhalifa**

### CODE REVIEW

File: wordcloud/wordcloud.py [hunk 1]
```diff
@@ -538,7 +540,13 @@ def generate_from_text(self, text):
         -------
         self
         """
-        words = self.process_text(text)
```
Comment (amueller) [inline wordcloud/wordcloud.py]: you don't need to change the wordcloud class. You can just use ``generate_from_frequencies``.

**Comment by mohammadKhalifa** [comment]
I know. But I didn't want to expose it in the wordcloud_cli.py. Do you think its okay?

File: wordcloud/wordcloud.py (hunk 1 above)
Comment (mohammadKhalifa) [inline wordcloud/wordcloud.py]: I know. But I didn't want to expose it in wordcloud_cli.py. Do you think its okay?

Comment (mohammadKhalifa) [inline wordcloud/wordcloud.py]: I know. But I didn't want to expose it in wordcloud_cli.py. Do you think its okay?

Comment (amueller) [inline wordcloud/wordcloud.py]: it's not internal, it's public and documented as such.

**Comment by mohammadKhalifa** [comment]
Okay. Will I need to add `ArgOption` in `test_wordcloud_cli.py` for my new argument  ?

**Comment by amueller** [comment]
Sorry I'm not sure I understand the question.

**Comment by mohammadKhalifa** [comment]
I have the following error in one of the build checks 
```
ERROR: test_wordcloud_cli.test_main_passes_arguments_through
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/amueller/word_cloud/test/test_wordcloud_cli.py", line 45, in test_main_passes_arguments_through
    cli.main(args)
  File "/home/travis/build/amueller/word_cloud/wordcloud/wordcloud_cli.py", line 20, in main
    weightedwords = args.weightedwords).generate(args.text)
AttributeError: 'Namespace' object has no attribute 'weightedwords'
```

AS far as I know, I need to add `weightedwords` to the arguments in `test_wordcloud_cli.py`

**Comment by mohammadKhalifa** [comment]
The problem with using `generate_from_frequencies` right away is that a user of the `WordClass` method will have to create the weights dict and pass it to the method themselves which would restrict the feature to users of the cli only. I believe I have to modify `WordCloud` to allow user to use this feature immediately by setting the `weightedwords` arg and passing the text to `generate`. Based on the argument `generate` creates the dict and passes it to `generate_from_frequencies`

**Comment by amueller** [comment]
Sorry I don't get your point. This feature is already available for users of the WordCloud class as ``generate_from_frequency``.

File: setup.py [hunk 2]
```diff
@@ -9,10 +9,11 @@
     url='https://github.com/amueller/word_cloud',
     description='A little word cloud generator',
     license='MIT',
-    install_requires=['matplotlib', 'numpy>=1.6.1', 'pillow'],
+    install_requires=['matplotlib', 'numpy>=1.6.1', 'pillow', 'python-bidi', 'arabic-reshaper'],
```
Comment (amueller) [inline setup.py]: These shouldn't be general dependencies.

**Comment by amueller** [comment]
this branch now seems to contain something else entirely?!

**Comment by mohammadKhalifa** [comment]
Yes I've been woking on Arabic language support and I commited to this PR by mistake. I will reset this branch.

**Comment by mohammadKhalifa** [comment]
@amueller  What I'm trying to say is that `generate_from_frequencies` requires a `dict`. I want to spare the user to have to create the dict so they could just pass the csv text to the `WordCloud` object and setting the `weightedword` argument to be true.

**Comment by amueller** [comment]
I think if people use the programmatic interface they can do the conversion themselves. Adding all possible interfaces to the library complicates the code with near zero benefit to the user.

**Pull request closed without merging**